src/services/codefixes/importFixes.ts

```diff
@@ -0,0 +1,279 @@
+                cancellationToken.throwIfCancellationRequested();
+
+                const exports = checker.getExportsOfModule(moduleSymbol) || [];
+                for (const exported of exports) {
+                    if (exported.name === name) {
+                        allActions.push(getCodeActionForImport(moduleSymbol));
+                    }
+                }
+            }
+
+            return allActions;
+
+            function getCodeActionForImport(moduleSymbol: Symbol): CodeAction {
+                // Check to see if there are already imports being made from this source in the current file
+                const existingDeclaration = forEach(sourceFile.imports, importModuleSpecifier => {
+                    const importSymbol = checker.getSymbolAtLocation(importModuleSpecifier);
+                    if (importSymbol === moduleSymbol) {
+                        return getImportDeclaration(importModuleSpecifier);
+                    }
+                });
+
+                if (existingDeclaration) {
+                    return getCodeActionForExistingImport(existingDeclaration);
+                }
+                else {
+                    return getCodeActionForNewImport();
+                }
+
+                function getImportDeclaration(moduleSpecifier: LiteralExpression) {
+                    let node: Node = moduleSpecifier;
+                    while (node) {
+                        if (node.kind !== SyntaxKind.ImportDeclaration) {
+                            node = node.parent;
+                        }
+
+                        return <ImportDeclaration>node;
+                    }
+                    return undefined;
+                }
+
+                function getCodeActionForExistingImport(declaration: ImportDeclaration): CodeAction {
+                    const moduleSpecifier = declaration.moduleSpecifier.getText();
+
+                    // We have to handle all of the different import declaration forms
+                    if (declaration.importClause) {
+                        if (declaration.importClause.namedBindings) {
+                            const namedBindings = declaration.importClause.namedBindings;
+                            if (namedBindings.kind === SyntaxKind.NamespaceImport) {
+                                /**
+                                 * Cases:
+                                 *     import * as ns from "mod"
+                                 *     import d, * as ns from "mod"
+                                 *
+                                 * Because there is no import list, we alter the reference to include the
+                                 * namespace instead of altering the import declaration. For example, "foo" would
+                                 * become "ns.foo"
+                                 */
+                                const ns = (<NamespaceImport>namedBindings).name.getText();
+                                return createCodeAction(

```

Comment:
After trying it I find it quite annoying if we insert a new line for every single imports. I tried webstorm, and they do consolidate imports from the same file together.